auth/client/iam/client.go

```diff
@@ -63,7 +70,14 @@ func (hb HTTPClient) OAuthAuthorizationServerMetadata(ctx context.Context, oauth
 	}
 	var metadata oauth.AuthorizationServerMetadata
 	if err = hb.doGet(ctx, metadataURL.String(), &metadata); err != nil {
-		return nil, err
+		// if this is a core.HttpError and the status code >= 500 then we want the caller to receive a 502 Bad Gateway
+		// we do this by changing the status code of the error
+		// any other error should result in a 400 Bad Request
+		if httpErr, ok := err.(core.HttpError); ok && httpErr.StatusCode >= 500 {
+			httpErr.StatusCode = http.StatusBadGateway
```

Comment:
untested; don't know how hard it is to test

Reply:
I selected code and asked CoPilot to generate the test for me....